conn/physic/units.go

```diff
@@ -848,18 +860,47 @@ func atod(s string) (decimal, int, error) {
 	seenDigit := false
 	seenZero := false
 	isPoint := false
+	seenPlus := false
 
 	// Strip leading zeros, +/- and mark DP.
 	for i := 0; i < len(s); i++ {
 		switch {
 		case s[i] == '-':
+			if seenPlus {
```

Comment:
I just realized; I think this string would pass: "1-2" because you don't deny if either seenDigit or seenZero is true. Same for case '+'. Please a test case to confirm.

Reply:
Yes. It is a bug. nicely spotted!
The question is should return a decimal 1 or error?
I'm thinking error. I'm having a hard time thinking of a case where it would not be.

Reply:
I'm ambivalent but somewhat leaning towards not an error;
- If it is an error, then it's a one-off compared to the letters, "10-" is invalid, but "10n" is valid.
- if it is parsed properly, parseSIUnits() will then error out properly which I think is better.